database/sqlite/sqlite_functions.c

```diff
@@ -2208,3 +2212,263 @@ void sql_load_node_id(RRDHOST *host)
 
     return;
 };
+
+/*
+ * Store a gap in the database
+ */
+int sql_store_gap(
+    uuid_t *gap_uuid,
+    char *host_mguid,
+    int t_delta_start,
+    int t_delta_first,
+    int t_delta_end,
+    char *status)
+{
+    sqlite3_stmt *res = NULL;
+    int rc, param = 0;
+
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode != RRD_MEMORY_MODE_DBENGINE)
+            return 0;
+        error_report("Database has not been initialized");
+        return SQLITE_ERROR;
+    }
+
+    if (unlikely(!res)) {
+        rc = prepare_statement(db_meta, SQL_STORE_GAP, &res);
+        if (unlikely(rc != SQLITE_OK)) {
+            error_report("Failed to prepare statement to store gap, rc = %d", rc);
+            return SQLITE_ERROR;
+        }
+    }
+
+    param++;
+    rc = sqlite3_bind_blob(res, 1, gap_uuid, sizeof(*gap_uuid), SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+    
+    param++;
+    rc = sqlite3_bind_text(res, 2, host_mguid, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+
+    param++;
+    rc = sqlite3_bind_int(res, 3, t_delta_start);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+    
+    param++;
+    rc = sqlite3_bind_int(res, 4, t_delta_first);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+    
+    param++;
+    rc = sqlite3_bind_int(res, 5, t_delta_end);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+    
+    param++;
+    rc = sqlite3_bind_text(res, 6, status, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+
+    rc = execute_insert(res);
+    if (unlikely(rc != SQLITE_DONE))
+        error_report("Failed to store gap, rc = %d", rc);
+
+    rc = sqlite3_reset(res);
```

Comment:
Please, also accept this [suggestion](https://github.com/netdata/netdata/pull/12556/files#r879024775).